src/webresource-handler/index.ts

```diff
@@ -81,22 +81,30 @@ const isFileInValidPath = async (
 		url: req.url,
 		method: req.method,
 	});
-	const resourceName = sbvrUtils.resolveSynonym(odataRequest);
+	const sqlResourceName = sbvrUtils.resolveSynonym(odataRequest);
+
+	const table = model.abstractSql.tables[sqlResourceName];
+
+	if (table == null) {
+		return false;
+	}
 
 	const permission = req.method === 'POST' ? 'create' : 'update';
 	const vocab = model.versions[model.versions.length - 1];
 
 	// Checks if it has permissions on both the original resourceName or any synonym
-	const hasPermissions =
-		(await checkPermissions(req, permission, resourceName, vocab)) ||
-		(await checkPermissions(req, permission, odataRequest.resourceName, vocab));
+	const hasPermissions = await checkPermissions(
+		req,
+		permission,
+		odataRequest.resourceName,
+		vocab,
+	);
 
 	if (!hasPermissions) {
 		return false;
 	}
 
-	const sqlResourceName = odataNameToSqlName(resourceName);
-	const fields = model.abstractSql.tables[sqlResourceName].fields;
+	const fields = table.fields;
```

Comment:
NIT: Maybe we just remove the whole fields variable and on the `.some` below we do 
```ts
return table.fields.some(...)
```

I did create the variable before because `model.abstractSql.tables[sqlResourceName].fields;` was a bit too long to inline, but `table.fields` should be ok and it is clear to read